src/components/implementation/cbuf_mgr/default/cbuf_mgr.c

```diff
@@ -643,8 +648,21 @@ cbuf_create(spdid_t spdid, unsigned long size, int cbid)
 		cbi->owner.spdid = spdid;
 		INIT_LIST(&cbi->owner, next, prev);
 		INIT_LIST(cbi, next, prev);
+	
+		if (cbi_head) {
+			struct cbuf_info *cbi_new = (struct cbuf_info *) malloc(sizeof(struct cbuf_info));
+			cbi_new->cbid = id;
+
+			ADD_END_LIST(cbi_head, cbi_new, next, prev);
+		}
+		else {
+			cbi_head = (struct cbuf_info *) malloc(sizeof(struct cbuf_info));
+			cbi_head->cbid = id;
+			
+			INIT_LIST(cbi_head, next, prev);
+		}
```

Comment:
I think it would be best to use this approach now (but I will move the two lines out like you suggest) and later on use a better approach that is not a global list.